### PR TITLE
[exporter/coralogixexporter] Add a list of AppName and SubSystem resource attributes

### PR DESCRIPTION
**Pull request by povilasv**
**Description:** <Describe what has changed.>

Adding a feature - allow users to map various Resource attributes to Coralogix Application Name / Subsystem fields.

**Link to tracking Issue:** #14398

**Testing:** 
- added more unit tests
- local testing

**Documentation:**

- updated README
- added various examples

### CODE REVIEW

File: exporter/coralogixexporter/README.md [hunk 1]
```diff
@@ -74,6 +83,100 @@ Depending on your region, you might need to use a different endpoint. Here are t
...
+    metrics:
+      endpoint: "otel-metrics.coralogix.com:443"
+    logs:
+      endpoint: "otel-logs.coralogix.com:443"
+    application_name_attributes:
+      - "applicationName" 
+    subsystem_name_attributes:
+      - "host.name"
+```
+
```
Comment (oded-dd) [inline exporter/coralogixexporter/README.md:179]: We should also add cloud attributes example (e.g. resourcedetection/ec2)

**Review by oded-dd — APPROVED** [review]

**Comment by oded-dd** [comment]
@codeboten can it please be addressed?

File: exporter/coralogixexporter/README.md [hunk 2]
```diff
@@ -74,6 +83,133 @@ Depending on your region, you might need to use a different endpoint. Here are t
 | EUROPE1 | `otel-traces.coralogix.com:443`     | `otel-metrics.coralogix.com:443`     | `otel-logs.coralogix.com:443`     |
 | EUROPE2 | `otel-traces.eu2.coralogix.com:443` | `otel-metrics.eu2.coralogix.com:443` | `otel-logs.eu2.coralogix.com:443` |
 
+### Application and SubSystem attributes
+
+v0.61 release of OpenTelemetry Collector allows you to map Application name and Subsystem name to Resource attributes.
```
Comment (codeboten) [inline exporter/coralogixexporter/README.md:88]:
```suggestion
v0.62.0 release of OpenTelemetry Collector allows you to map Application name and Subsystem name to Resource attributes. 
```

**Review by codeboten — COMMENTED** [review]
Just one suggestion, looks good otherwise

**Review by codeboten — APPROVED** [review]

**Pull request merged**